### ISSUE TITLE
python27Packages.faulthandler fails to build

### ISSUE THREAD

**Issue by stefano-m** [NixOS/nixpkgs#34415]
## Issue description

python27Packages.faulthandler fails to build on the unstable channel

Cc @sauyon ([faulthandler's maintainer](https://github.com/NixOS/nixpkgs/blob/804285f589ded484b530750dd1ec03b9052bcdac/pkgs/development/python-modules/faulthandler/default.nix#L16))

### Steps to reproduce

```shell
 nix-channel --list
unstable https://nixos.org/channels/nixpkgs-unstable
```

```shell
nix-channel --update unstable
downloading Nix expressions from ‘https://d3g5gsiof5omrk.cloudfront.net/nixpkgs/nixpkgs-18.03pre126020.931a0b8be80/nixexprs.tar.xz’...
downloading ‘https://d3g5gsiof5omrk.cloudfront.net/nixpkgs/nixpkgs-18.03pre126020.931a0b8be80/nixexprs.tar.xz’... [8271/9692 KiB, 813.3 KiB/s]
unpacking channels...
```

```shell
nix-env -iA unstable.python27Packages.faulthandler
```

The output shows test failures due to regular expressions not matching. E.g.

```
AssertionError: Regex didn't match: '^Fatal Python error: Segmentation fault\n\nCurrent\\ thread\\ XXX\\ \\<python\\>\\ \\(most\\ recent\\ call\\ first\\):\n  File "<string>", line 4 in <module>' not found in u'Fatal Python error: Segmentation fault\n\nCurrent thread XXX <python2.7> (most recent call first):\n  File "<string>", line 4 in <module>'
```
It seems that the tests are expecting to see `python` in the message, while the actual error says `python2.7`. If it's just that, I hope that it would be easy to fix the build.

I am happy to help if given some guidance.

Full output:
* https://gist.github.com/stefano-m/bf3f62b5597f79069adddf2d91b84c57

## Technical details

- system: `"x86_64-linux"`
 - host os: `Linux 4.9.78, NixOS, 17.09.2875.c2b668ee726 (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-17.09.2875.c2b668ee726"`
 - channels(user): `"unstable-18.03pre126020.931a0b8be80"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by dotlambda** [comment]
This is an upstream issue. See similar issue https://github.com/vstinner/faulthandler/issues/32.
Best write them that they should also accept `python2.7`, `python3.6`, etc.
Then you can open a PR here that adds a patch to the derivation.

**Comment by stefano-m** [comment]
Thanks, will look into that.

It probably makes sense to keep this issue open to track the upstream one though.

**Comment by stefano-m** [comment]
Note that a patch is available (https://github.com/vstinner/faulthandler/commit/67b661ebdbbf4067bd173af4d61ba815b9d49616 ), it may make sense to apply it (e.g. done for Fedora https://github.com/vstinner/faulthandler/issues/32#issuecomment-330528946) while waiting for a new release.

I am still happy to try and do that.

**Comment by dotlambda** [comment]
I don't think that patch will work. It accepts `python2` but not `python2.7`.

**Comment by stefano-m** [comment]
Sorry, I missed that :(

**Comment by dotlambda** [comment]
I opened an upstream PR: https://github.com/vstinner/faulthandler/pull/34
And I applied the patch in Nixpkgs: https://github.com/NixOS/nixpkgs/pull/35159

**Comment by stefano-m** [comment]
@FRidh fyi https://github.com/vstinner/faulthandler/pull/34 has now been merged and a release is pending (see https://github.com/vstinner/faulthandler/issues/32#issuecomment-383491557)... hopefully we won't need the patch after that.

Thanks everyone for your work!

**Comment by stefano-m** [comment]
https://github.com/vstinner/faulthandler/issues/32#issuecomment-383565754

> I released faulthandler 3.1. Building wheel packages on Windows (32 and 64 bits binaries for Python 2.7) is always annoying, but it's now done. Enjoy.

**Comment by dotlambda** [comment]
I removed the patches in 83301ffab03c86b291677b64b5cec7026f412cbda5ebd27e4cb3338452c40021.
